### ISSUE TITLE
Why can't a static field be accessed through an instance?

### ISSUE THREAD

**Issue by GoldenJoe** [dart-lang/sdk#35228]
Dart Version: 2.0.0
OS: OSX
IDE: VS Code

I'm making a simple class for managing colors:

`
class _Text {
  const _Text();
  final Color Gray5C = const Color(0x5c5c5c); // warning: [dart] Name non-constant identifiers using lowerCamelCase. [non_constant_identifier_names]
}

class MyColors {
  const MyColors._();
  static const Text = _Text();
}

class TestClass {
  var x = MyColors.Text.Gray5C;
}
`

With some help, I got the above working. However, one of the earlier solutions I tried was:

`
class _Text {
  const _Text();
  static const Gray5C = Color(0x5c5c5c);
}

class MyColors {
  const MyColors._();
  static const Text = _Text();
}

class TestClass {
  var x = MyColors.Text.Gray5C; // error: [dart] Static field 'Gray5C' can't be accessed through an instance. [instance_access_to_static_member]
}
`
I don't understand why a const in an instance of a const class cannot be accessed through an instance. Personally, I feel that the incorrect way looks much cleaner, and it follows naming conventions. Can someone ELI5?

**Comment by vsmenon** [comment]
Dart doesn't support dynamic/virtual dispatch of static members.  This part of their semantics.  E.g., if a subclass of `_Text` defined its own `Gray5C`, it'd override in the instance case and not in static case.
